lua/holohud2/add-ons/tacrp.lua

```diff
@@ -22,7 +22,7 @@ HOLOHUD2.hook.Add( "GetWeaponFiremode", "tacrp", function( weapon )
 
     if weapon:GetSafe() then return FIREMODE_SAFE end
 
-    if not weapon.Firemodes or #weapon.Firemodes <= 0 then return end
+    if weapon:GetFiremodeAmount() <= 0 then return end
```

Comment:
TacRP uses both `SWEP.Firemode` and `SWEP.Firemodes`, so checking `weapon.Firemodes` is insufficient to determine whether the weapon has a firemode or not.